### PR TITLE
GlobalExceptionHandler and data validation.

### PR DESCRIPTION
**Pull request by ipantazi**
Added `CustomGlobalExceptionHandler` class.
Added dependency `validation` to the `pom.xml`.
Added validation for DTO classes.

### CODE REVIEW

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -29,7 +30,7 @@ public List<BookDto> getAll() {
     }
 
     @GetMapping("/search")
-    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
+    public List<BookDto> searchBooks(@Validated BookSearchParametersDto searchParameters) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/controller/BookController.java]:
```suggestion
    public List<BookDto> searchBooks(@Valid BookSearchParametersDto searchParameters) {
```

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java [hunk 2]
```diff
@@ -40,14 +41,14 @@ public BookDto getBookById(@PathVariable Long id) {
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public BookDto createBook(@RequestBody CreateBookRequestDto bookRequestDto) {
+    public BookDto createBook(@RequestBody @Validated CreateBookRequestDto bookRequestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/controller/BookController.java]:
```suggestion
    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookRequestDto) {
```

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java [hunk 3]
```diff
@@ -40,14 +41,14 @@ public BookDto getBookById(@PathVariable Long id) {
...
         return bookService.save(bookRequestDto);
     }
 
     @PutMapping("/{id}")
     @ResponseStatus(HttpStatus.OK)
     public BookDto updateBook(@PathVariable Long id,
-                              @RequestBody CreateBookRequestDto bookRequestDto) {
+                              @RequestBody @Validated CreateBookRequestDto bookRequestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/onlinebookstore/controller/BookController.java]:
```suggestion
                              @RequestBody @Valid CreateBookRequestDto bookRequestDto) {
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by JJJazl — APPROVED** [review]

**Comment by JJJazl** [comment]
Please, pay attention that you have merge conflicts. Be careful and resolve it properly to avoid challenges with your next PRs

**Pull request merged**